microcosm_eventsource/immutable_event_factory.py

```diff
@@ -0,0 +1,81 @@
+"""
+Factory for immutable events.
+
+In extension of EventFactory, this class provides a native support of event updating container objects based
+on event types
+
+"""
+from collections import defaultdict
+from typing import Dict
+
+from microcosm_eventsource.factory import EventFactory
+
+
+container_common_mutator_registry: Dict[str, list] = defaultdict(list)
+container_event_specific_mutator_registry: Dict[str, list] = defaultdict(list)
+
+
+def register_container_mutator_common(event_type):
+    def decorator(func):
+        container_common_mutator_registry[event_type.__name__].append(func.__name__)
+        return func
+
+    return decorator
+
+
+def register_container_mutator_by_event_type(event_instance_type):
```

Comment:
Based on: https://github.com/globality-corp/juvenal/pull/1628/files#diff-e305f5c2e0000c70d1b8dceb78a7c19fR81

It would be nice to have:
```
 def register_container_mutator_by_event_type(event_instance_type: Optional[str] = None, event_instance_types: Optional[List[str]] = None):
```
or
```
 def register_container_mutator_by_event_type(event_instance_types: List[str]):
```

Reply:
Also agree with @afallou on combining the two decorators. The risk of key collision seems remote so not sure what the benefit of the data separation is when it leads to two very similar decorators being needed.

Reply:
Also agree. In my case I would say let the decorator register for all event types or a specific set of event types based on the type baed as an argument to the decorator.

Reply:
My preference in code is always for explicitness in code, so I am inclined to keep separate decorators.